### PR TITLE
Added test suite attributes

### PR DESCRIPTION
**Pull request by wachambo**
See Junit 10 XSD:
https://github.com/jenkinsci/xunit-plugin/blob/master/src/main/resources/org/jenkinsci/plugins/xunit/types/model/xsd/junit-10.xsd#L92

### CODE REVIEW

**Comment by sysradium** [comment]
I guess you should only append the arguments in the `__init__` signature not to break anyones code.

**Comment by wachambo** [comment]
Hi! I made the changes you proposed for backward compatibility. Is it ok now?

File: junit_xml/__init__.py [hunk 1]
```diff
@@ -291,9 +318,10 @@ def _clean_illegal_xml_chars(string_to_clean):
 class TestCase(object):
     """A JUnit test case with a result and possibly some stdout or stderr"""
 
-    def __init__(self, name, assertions=None, elapsed_sec=None,
-            timestamp=None, classname=None, status=None, category=None, file=None, line=None,
-            log=None, group=None, url=None, stdout=None, stderr=None):
+    def __init__(self, name, classname=None, elapsed_sec=None, stdout=None,
```
Comment (sysradium) [inline junit_xml/__init__.py]: Here you have also changed the signtature

File: junit_xml/__init__.py [hunk 2]
```diff
@@ -308,6 +336,7 @@ def __init__(self, name, assertions=None, elapsed_sec=None,
         self.stdout = stdout
         self.stderr = stderr
 
+        self.enable = True
```
Comment (sysradium) [inline junit_xml/__init__.py]: It seems that it is better to call it `is_enabled`

File: junit_xml/__init__.py [hunk 3]
```diff
@@ -317,6 +346,10 @@ def __init__(self, name, assertions=None, elapsed_sec=None,
         self.skipped_message = None
         self.skipped_output = None
 
+    def disable(self):
```
Comment (sysradium) [inline junit_xml/__init__.py]: It is not java, so it is not necessary to add setters/getters. If in the future you'd need  to add some logic you can easily do that converting `is_enabled` to a property. So I would drop that if that's was your concern

File: junit_xml/__init__.py [hunk 4]
```diff
@@ -342,6 +375,10 @@ def add_skipped_info(self, message=None, output=None):
         if output:
             self.skipped_output = output
 
+    def is_disabled(self):
```
Comment (sysradium) [inline junit_xml/__init__.py]: Same here, I would drop that in favour of just using `is_enabled` flag.

File: test_junit_xml.py [hunk 5]
```diff
@@ -73,8 +73,8 @@ def test_single_suite_no_test_cases(self):
 
         (ts, tcs) = serialize_and_read(
             TestSuite(
-                'test',
-                [],
+                name='test',
```
Comment (sysradium) [inline test_junit_xml.py]: As you can see after you changed the constructors signature you broke the tests already. That could be someones code :)

File: junit_xml/__init__.py [hunk 6]
```diff
@@ -134,12 +151,22 @@ def build_xml_doc(self, encoding=None):
...
+            stderr_element = ET.SubElement(xml_element, "system-err")
+            stderr_element.text = decode(self.stderr, encoding)
+
         # test cases
         for case in self.test_cases:
             test_case_attributes = dict()
             test_case_attributes['name'] = decode(case.name, encoding)
             if case.assertions:
-                test_case_attributes['assertions'] = decode(case.assertions, encoding)
+                test_case_attributes['assertions'] = "%d" % case.assertions
```
Comment (sysradium) [inline junit_xml/__init__.py]: Can you comment that change?

File: junit_xml/__init__.py (hunk 1 above)
Comment (wachambo) [inline junit_xml/__init__.py]: I changed back the signature as it was in #38  before my  #41 (where I broke the signature)
`     def __init__(self, name, classname=None, elapsed_sec=None, stdout=None,
stderr=None):`

Comment (sysradium) [inline junit_xml/__init__.py]: OK, sorry.

File: test_junit_xml.py (hunk 5 above)
Comment (wachambo) [inline test_junit_xml.py]: Here, I just add attributes explicitly in constructor signature. I think I don't break anyone's code :___(

Comment (sysradium) [inline test_junit_xml.py]: Off course not. My comment was about this: imagine you have a function:
```
def foo(a, b, c): ...

def test_foo():
     foo('1', '2', [])
```
If you change the signature of `foo` to `foo(a, c, b)` you break the client code and the test as well. And I thought that realised and fixed that problem just modifying the test:
```
def test_foo():
     foo(a='1', b='2', c=[])
```
So the test now works but still there are guys out there whose code broke :) So the failing test was a red flag here. If you did not - then it's fine.

File: test_junit_xml.py [hunk 7]
```diff
@@ -213,7 +213,7 @@ def test_attribute_time(self):
 
     def test_attribute_disable(self):
         tc = TestCase('Disabled-Test')
-        tc.disable()
+        tc.is_enabled=False
```
Comment (sysradium) [inline test_junit_xml.py]: PEP8 :) Should be spaces around `=`

File: test_junit_xml.py [hunk 8]
```diff
@@ -338,7 +338,7 @@ def test_init_stdout_stderr(self):
 
     def test_init_disable(self):
         tc = TestCase('Disabled-Test')
-        tc.disable()
+        tc.is_enabled=False
```
Comment (sysradium) [inline test_junit_xml.py]: PEP8 :) Should be spaces around =

**Pull request merged**